src/sentry/static/sentry/app/components/organizations/multipleEnvironmentSelector.jsx

```diff
@@ -80,6 +80,7 @@ class MultipleEnvironmentSelector extends React.PureComponent {
   replaceSelected(newSelection) {
     this.setState({selectedEnvs: new Set(newSelection)});
     this.props.onChange(newSelection);
+    this.doUpdate();
```

Comment:
@markstory Internal state of the dropdown was being updated, but change was never saved/propagated (the environment would still exist in URL + store)

Reply:
I found https://github.com/getsentry/sentry/commit/1aaab50d7a4fb95023a06cc003daf99ce78e1a36#diff-0c4ce644581f429b0c68084cf1f86a35 in the commit logs, and the parts about `list -> details` navigation having issues might be something to check again.

Reply:
Hmm, is it better to go back to the original logic where we clear environments on an explicit project change? It seems unlikely you'd want to keep an environment selected when you change your project(s) context

Reply:
Given the numerous sharp edges it might be simplest.

Reply:
@markstory 	Updated to clear envs when we change projects https://github.com/getsentry/sentry/pull/13334/commits/15ea6fd3a0591a19d05d74a873170f3faca07c4d